pkgpanda/build/__init__.py

```diff
@@ -1203,18 +1242,25 @@ def clean():
         # TODO(cmaloney): Run a wrapper which sources
         # /opt/mesosphere/environment then runs a build. Also should fix
         # ownership of /opt/mesosphere/packages/{pkg_id} post build.
-        cmd.run("package-builder", [
-            "/bin/bash",
-            "-o", "nounset",
-            "-o", "pipefail",
-            "-o", "errexit",
-            "/pkg/build"])
+        if is_windows:
+            # Note that the build directory is mounted in Windows due to a docker limitation that does not allow
+            # a file to be mounted.
+            cmd.run("package-builder", ["powershell.exe",
+                                        "-file",
+                                        PKG_DIR + "/build/" + build_script_file])
+        else:
+            cmd.run("package-builder", [
+                "/bin/bash",
+                "-o", "nounset",
+                "-o", "pipefail",
+                "-o", "errexit",
+                PKG_DIR + "/build"])
```

Comment:
docker on windows is different in too many ways to Linux where they actually cannot be quite the same. I really have tried to keep them as similar as possible but they cannot because one platfotrm is Linux and the other is windows. They have a different kernel and file system. they have a different version of docker. they kernel that drives docker is different. therefore we have to have differences. 
I have worked as hard as I can to minuimize these differences but without re-writing windows and docker I cannot do that.

Reply:
The only differences between the two platforms this piece of code is working around are 1) Docker on Windows can't mount files as volumes and 2) we run the different builder containers differently because they take their shell as an argument (bash vs. powershell). We can remove those differences, making this bit of code simpler by removing some platform checks. We should be able to run either builder container like so: `docker run --name=<name> <directory_mount_args> <build_script_filename>`. This would mean 1) always mounting directories on both platforms (I think that just requires a change to the code here, not any Dockerfiles) and then 2) adding the appropriate shell setup (`bash -o nounset...`, `powershell.exe -file`) to the `ENTRYPOINT` for the builder Dockerfiles. At that point, the containers should be interchangeable, at least from the perspective of `docker run`. The only platform-specific logic remaining would be the smallest amount that's necessary: choosing which container image we build and run, and which script we pass it.

Reply:
I think this is a nice to have, and can be addressed at a later date.